metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java

```diff
@@ -276,4 +278,46 @@ public WorkflowExecution createAndAddInQueueDepublishWorkflowExecution(MetisUser
     return orchestratorService
         .addWorkflowInQueueOfWorkflowExecutions(metisUser, datasetId, workflow, null, priority);
   }
+
+  /**
+   * Determines whether a depublication can be triggered at this point. This is the case lf:
+   * <ul>
+   *   <li>
+   *     No workflow is currently in progress, and
+   *   </li>
+   *   <li>
+   *     The dataset has the status 'published' (as opposed to 'depublished' or 'neither'), and
+   *   </li>
+   *   <li>
+   *     The records in the dataset are ready for viewing.
+   *   </li>
+   * </ul>
+   * @param metisUser The user performing this operation. Cannot be null.
+   * @param datasetId The ID of the dataset for which to retrieve the records. Cannot be null.
+   * @return Whether a depublication can be triggered.
+   * @throws GenericMetisException which can be one of:
+   * <ul>
+   * <li>{@link eu.europeana.metis.core.exceptions.NoDatasetFoundException} if the dataset
+   * identifier provided does not exist</li>
+   * <li>{@link eu.europeana.metis.exception.UserUnauthorizedException} if the user is not
+   * authenticated or authorized to perform this operation</li>
+   * </ul>
+   */
+  public boolean canTriggerDepublication(MetisUser metisUser, String datasetId)
+          throws GenericMetisException {
+
+    // Authorize.
+    authorizer.authorizeReadExistingDatasetById(metisUser, datasetId);
+
+    // If a workflow execution is currently in progress, we can't depublish.
+    if (orchestratorService.getRunningOrInQueueExecution(datasetId) != null) {
+      return false;
```

Comment:
Combine the two return types into one, using an if/else block

Reply:
Done.